### PR TITLE
[Data masking] Add codemod to apply `@unmask` to all named fragments

### LINKED ISSUE #11676: [Data masking] Provide codemod to add `@unmask` directives to all queries
Support:
* `gql` tags
* [`graphql` tags](https://the-guild.dev/graphql/codegen/plugins/presets/preset-client)
* `.graphql` files
* `.gql` files

### PR DESCRIPTION
**Pull request by jerelmiller**
Closes #11676

Adds a codemod that applies the `@unmask` directive to all named fragments in a file. This works for:

* `gql` (or `graphql`) tagged template literals
* `graphql` (or `gql`) functions
* `.graphql` or `.gql` files

To run the codemod, use the following:

```
npx jscodeshift -t scripts/codemods/data-masking/unmask.ts \
  --extensions tsx \
  --parser tsx \
  path/to/file.tsx
```

By default this adds the `@unmask` directive with no arguments. To add it in migrate mode, use the `--mode migrate` option in the CLI:

```
npx jscodeshift -t scripts/codemods/data-masking/unmask.ts \
  --extensions tsx \
  --parser tsx \
  path/to/file.tsx \
  --mode migrate
```

You can customize the tag name with the `--tag` option, or use it to only apply to a specific tag name. This defaults to both `gql` and `graphql` tags when not provided.

```
npx jscodeshift -t scripts/codemods/data-masking/unmask.ts \
  --extensions tsx \
  --parser tsx \
  path/to/file.tsx \
  --tag gql
```

Pass `--tag` multiple times to parse more than one tag name.

### CODE REVIEW

File: .changeset/flat-beans-knock.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+"@apollo/client": patch
+---
+
+Provide a codemod that applies `@unmask` to all named fragments for all operations and fragments. To use the codemod, run the following command:
+
+```
+npx jscodeshift -t node_modules/@apollo/client/scripts/codemods/data-masking/unmask.ts --extensions tsx path/to/app/
```
Comment (jerelmiller) [inline .changeset/flat-beans-knock.md:8]: I need to validate this is the correct path when the package is installed. So far I've been running this against the examples in this repo directly. If you'd like to do the same, the path to the `unmask.ts` script should be:

```
npx jscodeshift -t scripts/codemods/data-masking/unmask.ts --extensions tsx path/to/app/
```

Comment (phryneas) [reply .changeset/flat-beans-knock.md:8]: image

Scripts are not copied to `dist`, so this never gets published to npm. :/

File: .changeset/flat-beans-knock.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+```
```
Comment (phryneas) [inline .changeset/flat-beans-knock.md:9]:
```suggestion
```
npx jscodeshift -t node_modules/@apollo/client/scripts/codemods/data-masking/unmask.ts --extensions tsx --parser=tsx path/to/app/
```
```

**Comment by jerelmiller** [comment]
@phryneas 

> It would be great if it could change existing `@unmask` to `mode: "migrate"`

I considered that but left it for now because I thought that this might happen as a result of manually adding `@unmask` and figured its better to leave it alone than try and change it. Definitely one of those things though that have good arguments either way.

Should be easy to update though if we get feedback that it should update existing directives. If you feel strongly that we should have this out of the gate, I'd be happy to implement it.

**Comment by phryneas** [comment]
Please don't let that be a blocker! It would be a nice to have and I'm not even sure if it's very useful, so let's wait on that for user feedback!

**Comment by jerelmiller** [comment]
We still need to figure out the distribution strategy for the codemod, but that shouldn't hold us back from releasing the `rc`. Will merge now as we continue to explore that.

**Pull request merged**